### PR TITLE
[8] Add core database structure and payload handling and preparation

### PR DESCRIPTION
**Pull request by pawel-zielinski**

### CODE REVIEW

File: app/core/models.py [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
 
-# Create your models here.
+
+class User(DjangoUser):
+    profile = models.ForeignKey('Profile', on_delete=models.RESTRICT, related_name='profile')
+
+
+class Event(models.Model):
+    name = models.CharField(max_length=100)
+    organizer = models.ForeignKey(User, on_delete=models.DO_NOTHING, related_name='organizer')
```
Comment (Piotr-Zielinski-PZ) [inline app/core/models.py]: on_delete set to CASCADE

File: app/core/models.py [hunk 2]
```diff
@@ -1,3 +1,24 @@
 from django.db import models
+from django.contrib.auth.models import User as DjangoUser
 
-# Create your models here.
+
+class User(DjangoUser):
```
Comment (Piotr-Zielinski-PZ) [inline app/core/models.py:5]: for each model add Meta class with str representation and for Event model add ordering set to organize Event's objects with date

File: app/core/serializers.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        model = Event
+        fields = (
+            'name',
+            'organizer',
+            'tags',
+            'date',
+        )
+
+
+class EventSerializer(ModelSerializer):
```
Comment (Piotr-Zielinski-PZ) [inline app/core/serializers.py:17]: consider updating date to current date while update() and save()

File: app/core/serializers.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            'organizer',
+            'guests',
+            'tags',
+            'attachment',
+            'date',
+            'notes',
+        )
+
+
+class UserSerializer(ModelSerializer):
```
Comment (Piotr-Zielinski-PZ) [inline app/core/serializers.py:31]: set date_joined() to current date only while save()

Comment (Piotr-Zielinski-PZ) [reply app/core/serializers.py:31]: also exclude fields like is_staff from editing coz shit can get wild

File: app/core/models.py [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
+    organizer = models.ForeignKey(User, on_delete=models.DO_NOTHING, related_name='organizer')
+    guests = models.ManyToManyField(User, related_name='guests')
+    tags = models.ManyToManyField('Tag', related_name='tags')
+    date = models.DateTimeField()
+    attachment = models.FileField()
+    notes = models.CharField(max_length=250)
+
+
+class Profile(models.Model):
+    name = models.CharField(max_length=20)
```
Comment (Piotr-Zielinski-PZ) [inline app/core/models.py]: unique=True would be nice

File: app/core/serializers.py (hunk 3 above)
Comment (pawel-zielinski) [reply app/core/serializers.py:17]: obsolete

File: app/core/models.py (hunk 1 above)
Comment (pawel-zielinski) [reply app/core/models.py]: done

File: app/core/models.py (hunk 5 above)
Comment (pawel-zielinski) [reply app/core/models.py]: done

File: app/core/serializers.py (hunk 4 above)
Comment (pawel-zielinski) [reply app/core/serializers.py:31]: > set date_joined() to current date only while save()

why?

**Pull request merged**